### ISSUE TITLE
PHP 8.3/8.4 support

### ISSUE THREAD

**Issue by beryllium** [lildude/phpSmug#92]
Hello there, and thank you for writing this library! I'm a backend developer at SmugMug, and I've been finding this library useful for a tool I've been writing that talks to SmugMug's API.

I've noticed that Guzzle 6 has some internal logic that consistently triggers a noisy E_DEPRECATED notice in PHP 8.3/8.4.

I think the path forward would be to upgrade the library to use Guzzle 7.5+. I would also suggest updating composer.json to require at least PHP 8.1, and phpunit 10 or 11, as well as updating the way code sniffing is applied.

Would you be open to having me start some PRs for those changes? If you have another approach in mind, I'm happy to discuss things.

**Comment by lildude** [comment]
> Would you be open to having me start some PRs for those changes?

Definitely. I've been meaning to find some time to update things like this (and move dev into a container) but time is like hen's teeth these days.

**Comment by lildude** [comment]
A lot of dependencies have now been updated so things should hopefully work without issue on PHP 8.3 and 8.4 now.

**Comment by lildude** [comment]
Oh yes. I'll tag a new version later today.
